cmd/connector_celo.go

```diff
@@ -0,0 +1,100 @@
+//+build celo
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/celo-org/celo-blockchain/accounts/keystore"
+
+	"github.com/keep-network/keep-common/pkg/chain/celo/celoutil"
+	"github.com/keep-network/keep-ecdsa/config"
+	eth "github.com/keep-network/keep-ecdsa/pkg/chain"
+	"github.com/keep-network/keep-ecdsa/pkg/chain/celo"
+)
+
+// Values related with balance monitoring.
+//
+// defaultBalanceAlertThreshold determines the alert threshold below which
+// the alert should be triggered.
+var defaultBalanceAlertThreshold = big.NewInt(500000000000000000) // 0.5 CELO
+
+// defaultBalanceMonitoringTick determines how often the monitoring
+// check should be triggered.
+const defaultBalanceMonitoringTick = 10 * time.Minute
+
+func connectChain(
+	ctx context.Context,
+	config *config.Config,
+) (eth.Handle, *operatorKeys, error) {
+	celoKey, err := celoutil.DecryptKeyFile(
+		config.Celo.Account.KeyFile,
+		config.Celo.Account.KeyFilePassword,
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf(
+			"failed to read key file [%s]: [%v]",
+			config.Celo.Account.KeyFile,
+			err,
+		)
+	}
+
+	celoChain, err := celo.Connect(celoKey, &config.Celo)
+	if err != nil {
+		return nil, nil, fmt.Errorf(
+			"failed to connect to celo node: [%v]",
+			err,
+		)
+	}
+
+	// TODO: initialize Celo extensions
+	initializeBalanceMonitoring(ctx, config, celoChain, celoKey)
+
+	operatorKeys := &operatorKeys{
+		public:  &celoKey.PrivateKey.PublicKey,
+		private: celoKey.PrivateKey,
+	}
+
+	return celoChain, operatorKeys, nil
+}
+
+func initializeBalanceMonitoring(
+	ctx context.Context,
+	config *config.Config,
+	celoChain *celo.CeloChain,
+	celokey *keystore.Key,
+) {
+	balanceMonitor, err := celoChain.BalanceMonitor()
+	if err != nil {
+		logger.Errorf("error obtaining balance monitor handle [%v]", err)
+		return
+	}
+
+	alertThreshold := defaultBalanceAlertThreshold
+	if value := config.Celo.BalanceAlertThreshold; value != nil {
+		alertThreshold = value.Int
+	}
+
+	address := celokey.Address.Hex()
+
+	balanceMonitor.Observe(
+		ctx,
+		address,
+		alertThreshold,
+		defaultBalanceMonitoringTick,
+	)
```

Comment:
In [unchainded refactor](https://github.com/keep-network/keep-ecdsa/pull/693#discussion_r576783818) PR we reached the conclusion that the balance monitor can be Ethereum (or Ethereum-like) specific concept. If we get rid of it from `keep-core/pkg/chain/chain.go` interface, we can have it called in connectors with a specific currency passed to `Observe` function.

Reply:
Addressed in https://github.com/keep-network/keep-ecdsa/pull/686/commits/dd5a78dfd2b8fb27e1e14a3908966d560d23a949